File: servidor/fuente/RedDiff.Aplicacion/Auditorias/ModelosAuditorias.cs
namespace RedDiff.Aplicacion.Auditorias;

public sealed record AuditoriaResumen(
    long Id,
    long? UsuarioId,
    string? NombreUsuario,
    string Accion,
    string Entidad,
    long? EntidadId,
    DateTimeOffset Fecha,
    string Estado,
    string? Detalle);

public sealed record PaginaAuditorias(
    IReadOnlyList<AuditoriaResumen> Registros,
    int Pagina,
    int TamanoPagina,
    int TotalRegistros,
    int TotalPaginas);

public sealed record UsuarioFiltroAuditoria(
    long Id,
    string NombreUsuario);

public sealed record CatalogoFiltrosAuditoria(
    IReadOnlyList<UsuarioFiltroAuditoria> Usuarios,
    IReadOnlyList<string> Acciones,
    IReadOnlyList<string> Entidades);

public sealed record FiltrosAuditorias(
    long? UsuarioId,
    string? Accion,
    string? Entidad,
    string? Estado,
    DateTimeOffset? Desde,
    DateTimeOffset? Hasta,
    int Pagina = 1,
    int TamanoPagina = 25);

File: cliente/src/app/modulos/auditorias/modelos-auditorias.ts
export type EstadoAuditoria = 'Exitoso' | 'Fallido';

export interface AuditoriaResumen {
  id: number;
  usuarioId: number | null;
  nombreUsuario: string | null;
  accion: string;
  entidad: string;
  entidadId: number | null;
  fecha: string;
  estado: EstadoAuditoria;
  detalle: string | null;
}

export interface ResultadoPaginaAuditorias {
  registros: AuditoriaResumen[];
  pagina: number;
  tamanoPagina: number;
  totalRegistros: number;
  totalPaginas: number;
}

export interface UsuarioFiltroAuditoria {
  id: number;
  nombreUsuario: string;
}

export interface CatalogoFiltrosAuditoria {
  usuarios: UsuarioFiltroAuditoria[];
  acciones: string[];
  entidades: string[];
}

export interface FiltrosAuditorias {
  usuarioId?: number;
  accion?: string;
  entidad?: string;
  estado?: EstadoAuditoria;
  desde?: string;
  hasta?: string;
  pagina: number;
  tamanoPagina: number;
}
